common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -5273,6 +5274,21 @@ public void logVars(PrintStream ps) {
     }
   }
 
+  /**
+   * @return a ZooKeeperHiveHelper instance containing the ZooKeeper specifications from the
+   * given HiveConf.
+   */
+  public ZooKeeperHiveHelper getZKConfig() {
+    return new ZooKeeperHiveHelper(getVar(HiveConf.ConfVars.HIVE_ZOOKEEPER_QUORUM),
+            getVar(HiveConf.ConfVars.HIVE_ZOOKEEPER_CLIENT_PORT),
+            getVar(HiveConf.ConfVars.HIVE_SERVER2_ZOOKEEPER_NAMESPACE),
+            (int) getTimeVar(HiveConf.ConfVars.HIVE_ZOOKEEPER_SESSION_TIMEOUT,
```

Comment:
In the original code, this cast was needed because Builder#sessionTimeoutMs() accepts an int arguement. In the refactored code, I have retained that cast. We will need this cast (and have the risk of data conversion loss) here or at the time of invoking Builder#sessionTimeoutMs(). I thought it better to convert here to keep the values in ZooKeeperHiveHelper class to be in sync with the values passed to Builder#sessionTimeoutMs().